docs/pages/setup/guides/terraform-provider/reference.mdx

```diff
@@ -0,0 +1,422 @@
+---
+title: Terraform Provider reference
+description: Terraform Provider reference
+---
+
+Supported resources:
+
+* [`teleport_user`](#teleport_user)
+* [`teleport_role`](#teleport_role)
+* [`teleport_provision_token`](#teleport_provision_token)
+* [`teleport_github_connector`](#teleport_github_connector)
+* [`teleport_oidc_connector`](#teleport_oidc_connector)
+* [`teleport_saml_connector`](#teleport_saml_connector)
+* [`teleport_trusted_cluster`](#teleport_trusted_cluster)
+
+## Connecting the provider
+
+Add the following configuration section into your `terraform` configuration block:
+
+```
+terraform {
+  required_providers {
+    teleport = {
+      version = "0.0.1"
+      source  = "gravitational.com/teleport/teleport"
+    }
+  }
+}
+```
+
+## Provider configuration
+
+Provider supports following options in it's base configuration:
+
+| Name                    | Type       | Description                                               |
+|-------------------------|------------|-----------------------------------------------------------|
+|                   `addr`|     string | Teleport auth proxy host:port                             |
+|              `cert_path`|     string | Path to Teleport certificate file                         |
+|     `identity_file_path`|     string | Path to Teleport identity file                            |
+|               `key_path`|     string | Path to Teleport key file                                 |
+|            `profile_dir`|     string | Teleport profile path                                     |
+|           `profile_name`|     string | Teleport profile name                                     |
+|           `root_ca_path`|     string | Path to Teleport CA file                                  |
+
+You need to specify either:
+
+* `cert_path`, `key_path`,`root_ca_path` and `addr` to connect using key files.
+* `identity_file_path` and `addr` to connect using identity file.
+* `profile_name` and `profile_dir` (both can be empty) and Teleport will try to connect using current profile from `~/.tsh`
+
+Example:
+
+```
+provider "teleport" {
+  addr         = "localhost:3025"
+  cert_path    = "tf.crt"
+  key_path     = "tf.key"
+  root_ca_path = "tf.ca"
+}
+```
+
+# <a name="teleport_user"></a>`teleport_user`
```

Comment:
Great job splitting up each of these sections! Super easy to read through and find what you need.

Nitpicks:

1. Can you remove the anchor tags? Links to the H1, H2, ... headings will be automatically generated.
2. Can you remove tic marks from the headings?